cmd/wallet-sdk-gomobile/docs/usage.md

```diff
@@ -578,31 +658,147 @@ let issuerURI = interaction.issuerURI() // Optional (but useful)
 // Consider checking the activity log at some point after the interaction
 ```
 
+#### Kotlin (Android) - Authorization Code Flow
+
+```kotlin
+import dev.trustbloc.wallet.sdk.api.*
+import dev.trustbloc.wallet.sdk.localkms.Localkms
+import dev.trustbloc.wallet.sdk.localkms.MemKMSStore
+import dev.trustbloc.wallet.sdk.did.Resolver
+import dev.trustbloc.wallet.sdk.did.Creator
+import dev.trustbloc.wallet.sdk.openid4ci.*
+import dev.trustbloc.wallet.sdk.openid4ci.Opts
+import dev.trustbloc.wallet.sdk.verifiable.CredentialsArray
+
+// Setup
+val memKMSStore = MemKMSStore.MemKMSStore()
+val kms = Localkms.newKMS(memKMSStore)
+val didResolver = Resolver(null)
+val didCreator = Creator(kms as KeyWriter)
+val didDocResolution = didCreator.create("key", null) // Create a did:key doc
+
+val activityLogger = mem.ActivityLogger()
+
+// Going through the flow
+val args = Args("YourRequestURIHere", kms.getCrypto(), didResolver)
+val opts = Opts().setActivityLogger(activityLogger) // Optional, but useful for tracking credential activity
+val interaction = Interaction(args, opts) // This is a stateful object - we will use this to go through the flow.
+
+// It's a good idea to check the issuer's capabilities first
+val issuerCapabilities = interaction.issuerCapabilities()
+if (!issuerCapabilities.AuthorizationCodeGrantTypeSupported()) {
+    // This code example isn't applicable. See the pre-authorized code flow example instead.
+    
+    return
+}
+
+val scopes = StringArray()
+scopes.append("scope1").append("scope2")
+
+// If scopes aren't needed, call interaction.createAuthorizationURL() instead.
+val authorizationLink := interaction.createAuthorizationURLWithScopes("clientID", "redirect URI", scopes)
+
+// Open authorizationLink in a browser. Once the user has finished logging in, call requestAccessToken()
+// with the full redirect URI (including query parameters) that the login service sent the user to.
+// The code below assumes this has already been done somehow and that the URI is in the redirectURIWithParams variable.
+// In actual code, the call to requestAccessToken() couldn't be immediately after the createAuthorizationURLWithScopes()
+// call like in this example since control has to flow back to the user first.
+val redirectURIWithParams = "Put the redirect URI with params here"
+
+// If no error occurred, then authorization was successful. The access token will be stored in the interaction object
+// automatically.
+interaction.requestAccessToken(redirectURIWithParams)
+
+val credentials = interaction.requestCredential(didDocResolution.assertionMethod())
```

Comment:
Is it possible to combine above two methods into one method ? Something like `interaction.requestCredentialWithAuth(didDocResolution.assertionMethod(), redirectURIWithParams)`.

Reply:
Good idea. Done.